ui-admin/src/study/kits/KitEnrolleeSelection.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useEffect, useState } from 'react'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faCheck } from '@fortawesome/free-solid-svg-icons'
+import { Link } from 'react-router-dom'
+import {
+  ColumnDef,
+  ColumnFiltersState,
+  getCoreRowModel,
+  getFilteredRowModel,
+  getSortedRowModel,
+  useReactTable,
+  VisibilityState
+} from '@tanstack/react-table'
+
+import { Survey } from '@juniper/ui-core/build/types/forms'
+
+import Api, { Enrollee } from 'api/api'
+import { StudyEnvContextT } from 'study/StudyEnvironmentRouter'
+import { basicTableLayout, ColumnVisibilityControl, IndeterminateCheckbox } from 'util/tableUtils'
+import LoadingSpinner from 'util/LoadingSpinner'
+import RequestKitModal from '../participants/RequestKitModal'
+
+/**
+ * Interface for filtering/selecting enrollees who should receive sample kits.
+ */
+export default function KitEnrolleeSelection({ studyEnvContext }: { studyEnvContext: StudyEnvContextT }) {
+  const { portal, study, currentEnv, currentEnvPath } = studyEnvContext
+  const [isLoading, setIsLoading] = useState(true)
+  const [enrollees, setEnrollees] = useState<Enrollee[]>([])
+  const [rowSelection, setRowSelection] = useState<Record<string, boolean>>({})
+  const [columnVisibility, setColumnVisibility] = useState<VisibilityState>({})
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([])
+  const [showRequestKitModal, setShowRequestKitModal] = useState(false)
+
+  const loadEnrollees = async () => {
+    setIsLoading(true)
+    const enrollees = await Api.fetchEnrollesForKitManagement(
+      portal.shortcode, study.shortcode, currentEnv.environmentName)
+    setEnrollees(enrollees)
+    setIsLoading(false)
+  }
+
+  useEffect(() => {
+    loadEnrollees()
+  }, [])
+
+  const onSubmit = async (kitType: string) => {
+    const enrolleesSelected = Object.keys(rowSelection)
+      .filter(key => rowSelection[key])
+      .map(key => enrollees[parseInt(key)].shortcode)
+
+    // This iteration should be happening server-side: JN-460
+    for (const shortcode of enrolleesSelected) {
+      await Api.createKitRequest(
+        portal.shortcode, study.shortcode, currentEnv.environmentName, shortcode, kitType)
+    }
+
+    setShowRequestKitModal(false)
+    loadEnrollees()
+  }
+
+  const numSelected = Object.keys(rowSelection).length
+  const enableActionButtons = numSelected > 0
+
+  const hasEnrolleeCompletedSurvey = (enrollee: Enrollee, survey: Survey): boolean => {
```

Comment:
Some calculation can be done ahead of time, but still need to use `accessorFn` to do a looking by survey since the list of surveys is dynamic based on the study.